FirebaseRemoteConfig/Sources/RCNConfigSettings.m

```diff
@@ -427,7 +462,26 @@ - (BOOL)hasMinimumFetchIntervalElapsed:(NSTimeInterval)minimumFetchInterval {
 
   // Check if last config fetch is within minimum fetch interval in seconds.
   NSTimeInterval diffInSeconds = [[NSDate date] timeIntervalSince1970] - self.lastFetchTimeInterval;
-  return diffInSeconds > minimumFetchInterval;
+  BOOL hasMinimumFetchIntervalElapsed = (diffInSeconds > minimumFetchInterval) ? YES : NO;
+
+  // Check if we have a shared suite with an extension/app.
+  if (!hasMinimumFetchIntervalElapsed && _sharedUserDefaults) {
+    // Check if the latest eTag across main app and extensions is different than our eTag. If so,
+    // let the fetch go through.
+    NSString *allAppExtensionsLatestETag =
+        [_sharedUserDefaults objectForKey:RCNExternalUserDefaultsKeyLatestETag];
+
+    // If the eTag is different, check the fetch times.
+    if (![self.lastETag isEqualToString:allAppExtensionsLatestETag]) {
+      NSNumber *latestFetchTimeAcrossAppsAndExtensions =
+          [_sharedUserDefaults objectForKey:RCNExternalUserDefaultsKeyLastSuccessfulFetchTime];
```

Comment:
To confirm, if this returns `nil`, `doubleValue` should return `0` and `self.lastFetchTimeInterval` won't be set, right?

Reply:
Correct. It is a no-op as long as latestFetchTimeAcrossAppsAndExtensions is same or less than the current instance's latest fetch time.